compiler/llvm/clangUtil.cpp

```diff
@@ -2935,6 +2939,8 @@ static void helpComputeClangArgs(std::string& clangCC,
   // of cabs but it appears to slow down simple complex multiplication.
   if (ffloatOpt > 0) { // --no-ieee-float
     clangCCArgs.push_back("-ffast-math");
+    // turn off inf warnings
+    clangCCArgs.push_back("-Wno-nan-infinity-disabled");
```

Comment:
can you describe more about what problem this is solving? it's not obvious to me. (PR message might be a good place for such description)

Reply:
This silences warnings from `isinf`, `isfinite`, and `isnan` in `chplmath.h` when `--no-ieee-float` is used.

For example:
```
runtime/include/chplmath.h:33:70: warning: use of NaN is undefined behavior due to the currently enabled floating-point options [-Wnan-infinity-disabled]
   33 | MAYBE_GPU static inline int chpl_macro_float_isnan(float x) { return isnan(x); }
      |                                                                      ^~~~~~~~
/usr/include/math.h:930:20: note: expanded from macro 'isnan'
  930 | #  define isnan(x) __builtin_isnan (x)
```

This warning is caused by the usage of `-ffast-math`